osg_configure/configure_modules/gratia.py

```diff
@@ -358,20 +357,20 @@ def _make_subscription(self, probe, probe_file, probe_host, site, hostname):
             probe = 'gridftp-transfer'
 
         try:
-            buf = open(probe_file).read()
+            buf = open(probe_file, "r", encoding="latin-1").read()
             buf = self.replace_setting(buf, 'ProbeName', "%s:%s" % (probe, hostname))
             buf = self.replace_setting(buf, 'SiteName', site)
             buf = self.replace_setting(buf, 'Grid', self.grid_group)
             buf = self.replace_setting(buf, 'EnableProbe', '1')
             for var in ['SSLHost', 'SOAPHost', 'SSLRegistrationHost', 'CollectorHost']:
                 buf = self.replace_setting(buf, var, probe_host)
 
-            if not utilities.atomic_write(probe_file, buf, mode=420):
+            if not utilities.atomic_write(probe_file, buf, mode=0o644):
                 self.log("Error while configuring gratia probes: " +
                          "can't write to %s" % probe_file,
                          level=logging.ERROR)
                 raise exceptions.ConfigureError("Error configuring gratia")
-        except(IOError, OSError):
+        except OSError:
```

Comment:
I take it `IOError` just doesn't get raised in the above `try` block here, in python3?

Reply:
In Python 3, IOError and EnvironmentError are both aliases of OSError.